### DISCUSSION TITLE
Audio output format, bits per sample less relevant than sample rate for quality

### DISCUSSION THREAD

**Discussion by CircuitCM** [neonbjb/tortoise-tts#561] [Ideas]
Hi,

I noticed that the bit depth of the audio output is 32. From my experience the difference between the quality of a 32 bit sample and a 16 bit sample is very small, it has much less impact than the difference between a 44 HZ and 24 HZ sample frequency. But if it was possible to generate output with 16 bit depth that should increase the output rate by ~2x right? I suspect this can't be done because of how the model is built and trained?

**Comment by neonbjb** [comment]
You can perform inference in fp16 for these models and it does speed things up on server class hardware. On consumer hardware there is no change. The model does not actually output 32 bit audio since it never recorded anything it's just mimicking the dataset and it's actual output quality is probably worse than both 16bit real recordings.

**Reply by CircuitCM** [reply]
I see thanks. Then I'm guessing that the model's output is a continuous approximation of the waveform at each frame, and its training set is 24 HZ?
